app/views/restaurants/_form.html.erb

```diff
@@ -39,12 +39,12 @@
   <hr>
   <h3>Add Menu Items to Try</h3>
 
-  <% 3.times do %>
+  <div class="menu_item_fields">
     <%= f.fields_for :foods, @restaurant.foods.build do |g| %>
       <%= g.hidden_field :user_id, :value => current_user.id %>
       <div class="field">
         <%= g.label :name %>
-        <%= g.text_field :name %>
+        <%= g.text_field :name %> <a href="#" class="add_menu_item_field">+</a>
```

Comment:
I think you might want to make this a button instead of an anchor tag, as it is not creating a link. Also it might be better to not but this in the div.field tags.